parquet-hadoop/src/main/java/parquet/hadoop/InternalParquetRecordReader.java

```diff
@@ -202,12 +205,29 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
 
     while (!recordFound) {
       // no more records left
-      if (current >= total) { return false; }
+      if (current >= total) { return false;}
 
       try {
         checkRead();
-        currentValue = recordReader.read();
+
+        corruptRecordCounter.incRecordsSeen();
+        boolean corrupt = false;
+
+        try {
```

Comment:
I agree this loop is complicated. Unfortunately, the checkRead() method does a lot of internal state mutation, so I'm hesitant to add another layer around this class if that layer has to interact with the state that checkRead() does.

But I agree, this whole loop / class could use some cleanup. Actually some of the checks (like supporting 2 kinds of filters) could potentially go away once we are happy w/ the filter2 API.